editor/components/post-taxonomies/hierarchical-term-selector.js

```diff
@@ -287,4 +288,4 @@ export default connect(
 			return editPost( { [ restBase ]: terms } );
 		},
 	}
-)( withInstanceId( HierarchicalTermSelector ) );
+)( withSpokenMessages( withInstanceId( HierarchicalTermSelector ) ) );
```

Comment:
Minor: At this point we might consider introducing the `compose` utility to improve readability here:

```js
import { compose } from '@wordpress/components';

// ...

const applyConnect = connect(
 	( state, onwProps ) => {
 		return {
 			terms: getEditedPostAttribute( state, onwProps.restBase ),
 		};
 	},
 	{
 		onUpdateTerms( terms, restBase ) {
  			return editPost( { [ restBase ]: terms } );
  		},
  	}
);

export default compose(
	applyConnect,
	withSpokenMessages,
	withInstanceId
)( HierarchicalTermSelector );
```

Reply:
The only reference I see to compose is from `'@wordpress/element'` not `'@wordpress/components'` and it's basically `flowRight`, right? I see similar cases use directly `flowRight` or `flow` so which should I use?

Reply:
>The only reference I see to compose is from '@wordpress/element' not '@wordpress/components'

Correct, I was mistaken.

>and it's basically flowRight, right?

Yes. I think the original reasoning was that if we're emphasizing this pattern of higher-order components, it should be made a first-class part of the element abstraction.

>I see similar cases use directly flowRight or flow so which should I use?

Since #3907 all instances should have been updated to use `compose`.

If you're curious, there's a bit of detail at https://github.com/WordPress/gutenberg/pull/2500#discussion_r134669757 which explains the difference between `flow` and `flowRight` in the context of higher-order components.